src/Console/Diagnostic/SourceCodeIntegrityCommand.php

```diff
@@ -0,0 +1,164 @@
+        $output_buffer = fopen('php://memory', 'r+b');
+        $temp_output = new StreamOutput($output_buffer);
+        $summary = $checker->getSummary();
+        if (isset($summary['error'])) {
+            $output->writeln('<error>' . $summary['error'] . '</error>');
+            return 1;
+        }
+        $all_ok = true;
+        if (count(array_filter($summary, static fn($status) => $status === SourceCodeIntegrityChecker::STATUS_ALTERED)) > 0) {
+            $all_ok = false;
+        }
+        if (count(array_filter($summary, static fn($status) => $status === SourceCodeIntegrityChecker::STATUS_MISSING)) > 0) {
+            $all_ok = false;
+        }
+        if (count(array_filter($summary, static fn($status) => $status === SourceCodeIntegrityChecker::STATUS_ADDED)) > 0) {
+            $all_ok = false;
+        }
+        if ($all_ok) {
+            $temp_output->writeln('<info>' . __('GLPI source code integrity is validated.') . '</info>');
+        } else {
+            $temp_output->writeln('<error>' . __('GLPI source code integrity is not validated.') . '</error>');
+            $table = new Table($temp_output);
+            $table->setHeaders(['File', 'Status']);
+            foreach ($summary as $file => $status) {
+                $status_label = match ($status) {
+                    SourceCodeIntegrityChecker::STATUS_ALTERED => 'Altered',
+                    SourceCodeIntegrityChecker::STATUS_MISSING => 'Missing',
+                    SourceCodeIntegrityChecker::STATUS_ADDED => 'Added',
+                };
+                $table->addRow([
+                    $file,
+                    $status_label
+                ]);
+            }
+            $table->render();
+        }
+        if (!$diff) {
+            // copy lines from the temporary output buffer to the real output
+            rewind($output_buffer);
+            while (!feof($output_buffer)) {
+                $output->write(fread($output_buffer, 4096));
+            }
+        } else {
+            $errors = [];
+            if (VersionParser::isDevVersion(GLPI_VERSION)) {
+                $output->writeln('<error>' . __('Cannot generate a diff on a development version.') . '</error>');
+            } else {
+                $diff = '';
+                // enumerate lines of the temporary output buffer and add them to the diff prefixed with a # to make them comments
+                rewind($output_buffer);
+                while (!feof($output_buffer)) {
+                    $diff .= '# ' . fgets($output_buffer);
+                }
+                $diff .= "\n" . $checker->getDiff($errors, $allow_download);
+                if (count($errors) > 0) {
+                    $output->writeln('<error>' . __('Errors:') . '</error>');
+                    foreach ($errors as $error) {
+                        $output->writeln('<error>' . $error . '</error>');
+                    }
+                    return 1;
+                }
```

Comment:
If there is an error on a unique file, we should still display the diff we were able to compute.
Moving these lines after `$output->write(OutputFormatter::escape($diff));` would permit to do this, and to ensure that errors are still visible in the result.